bin/finalize_condition_entities.py

```diff
@@ -59,28 +61,30 @@ def process_weights(subset):
 
 args = parse_args()
 
-if not args.entrez_microbiomes_entities and not args.nucl_microbiomes_entities:
-    sys.exit("Neither --entrez-microbiome-entities nor --nucl-microbiome-entities were specified. Aborting.")
+if not args.entrez_conditions_entities and not args.nucl_conditions_entities:
+    sys.exit("Neither --entrez-conditions-entities nor --nucl-conditions-entities were specified. Aborting.")
 
 # Read and join the tables that provide microbiome_id, entity_id and entity_name
-entity_microbiome = pd.read_csv(args.microbiomes_entities_noweights, sep='\t')
+entity_microbiome = pd.read_csv(args.microbiomes_entities, sep='\t')
 entity            = pd.read_csv(args.entities, sep='\t')
 
-entity_microbiome = entity_microbiome.merge(entity)
+conditions        = pd.read_csv(args.conditions, sep="\t")
+
+entity_condition = entity_microbiome.merge(entity).merge(conditions)
 
 # Read the tables that provide the weights and concatenate them
-input_data = pd.concat([ pd.read_csv(e, sep='\t') for e in [args.entrez_microbiomes_entities, args.nucl_microbiomes_entities] if e ])
+input_data = pd.concat([ pd.read_csv(e, sep='\t', dtype={"entity_name":str}) for e in [args.entrez_conditions_entities, args.nucl_conditions_entities] if e ])
 
 # Join the weights against the entity ids table, which contains all entities
-# that we have observed in upstream processes. Thus, per microbiome, we expect
+# that we have observed in upstream processes. Thus, per condition, we expect
 # to find weights either for all of them or for none of them.
-result = entity_microbiome.merge(input_data, how="left").drop(columns="entity_name")
+result = entity_condition.merge(input_data, how="left").drop(columns="entity_name")
 
-# For each microbiome, we now check whether this assumption is true. If we find
-# no weights for a microbiome, we add uniform weights.
+# For each condition, we now check whether this assumption is true. If we find
+# no weights for a condition, we add uniform weights.
 try:
-    result = result.groupby("microbiome_id")\
+    result = result.groupby("condition_id")\
```

Comment:
I think the `process_weights` function will need adaptions for grouping via `condition_id` rather than `microbiome_id`

Reply:
Ah yes, I forgot to replace `microbiome_id` by `condition_id` in the `PartialWeightsError`. Or do you mean more changes? Because the functionality should still be the same, for one condition, we'll have all the weights or none, right?